components/engine_traits/src/range_cache_engine.rs

```diff
@@ -28,6 +28,9 @@ pub enum RegionEvent {
         source: CacheRegion,
         new_regions: Vec<CacheRegion>,
     },
+    Load {
```

Comment:
This  naming is vague. Maybe better merge `Load` and `Eviction` to `RoleChange` and let IME to decide how to handle this role change. Thus, we can even avoiding adding the new `LoadObserver`